src/prepared_statement.rs

```diff
@@ -0,0 +1,174 @@
+    }
+}
+
+impl PreparedStatement {
+    /// `index` represent the position of  marker `?` in the query
+    pub fn set_string(&mut self, index: u32, val: &'static str) -> error::Result<()> {
+        self.query_markers.insert(index, val.to_string().into());
+        Ok(())
+    }
+
+    // /// `index` represent the position of  marker `?` in the query
+    // /// date should be in the format epochtime
+    // not yet working
+    // pub fn set_date(&mut self, index: u32, val: i64) -> error::Result<()> {
+    //     let mut wtr = vec![];
+    //     match wtr.write_i64::<BigEndian>(val) {
+    //         Ok(_) => {
+    //             self.query_markers.insert(index, Value::new_normal(wtr.clone()));
+    //             Ok(())
+    //         }
+    //         Err(err) => return Err(error::Error::from(err)),
+    //     }
+    // }
+
+    /// `index` represent the position of marker `?` in the query
+    pub fn set_int(&mut self, index: u32, val: i32) -> error::Result<()> {
+        let mut wtr = vec![];
+        match wtr.write_i32::<BigEndian>(val) {
+            Ok(_) => {
+                self.query_markers.insert(index, Value::new_normal(wtr.clone()));
+                Ok(())
+            }
+            Err(err) => return Err(error::Error::from(err)),
+        }
+    }
+}
+
+/// This trait once imported into scope would provide the end users to set the values
+/// Prepare and execute statement with prepared Statement
+///
+
+pub trait PrepareAndExecute {
+    /// this returns a prepared statement
+    /// you can call execute_statement after setting the various parameters
+    fn prepare_statement(&mut self,
+                         query: String,
+                         with_tracing: bool,
+                         with_warnings: bool)
+                         -> error::Result<PreparedStatement>;
+
+
+    fn execute_statement(&mut self, prepared_statement: PreparedStatement) -> error::Result<Frame>;
+}
+
+
+impl<'a, T: Authenticator + 'a, X: CDRSTransport + 'a> PrepareAndExecute for Session<T, X> {
+    fn prepare_statement(&mut self,
+                         query: String,
+                         with_tracing: bool,
+                         with_warnings: bool)
+                         -> error::Result<PreparedStatement> {
+        let mut flags = vec![];
+        if with_tracing {
+            flags.push(Flag::Tracing);
+        }
+        if with_warnings {
+            flags.push(Flag::Warning);
+        }
+
+        let options_frame = Frame::new_req_prepare(query, flags).into_cbytes();
+
+        (self.cdrs.transport.write(options_frame.as_slice()))?;
+
+        parse_frame(&mut self.cdrs.transport, &self.compressor).map(|frame| {
+            PreparedStatement {
+                consistency: Consistency::One,
+                frame: Some(frame),
+                ..Default::default()
+            }
+        })
+    }
+
+    fn execute_statement(&mut self, prepared_statement: PreparedStatement) -> error::Result<Frame> {
```

Comment:
Now the bottle neck is the lack of implementators of `Into<Value>` for some of Rust types. But I'm working on that in task #81